testsuite/feelcrb/test_pbdw.cpp

```diff
@@ -0,0 +1,375 @@
+#define BOOST_TEST_MODULE pbdw testsuite
+
+#include <feel/feelcore/testsuite.hpp>
+#define USE_BOOST_TEST 1
+
+#include <feel/feelfilters/exporter.hpp>
+#include <feel/feelcrb/pbdw.hpp>
+#include <feel/feeldiscr/pch.hpp>
+#include <feel/feelfilters/loadmesh.hpp>
+#include <feel/feelvf/vf.hpp>
+
+using namespace Feel;
+
+inline
+po::options_description makeOptions()
+{
+    po::options_description options( "Test PBDW Options" );
+
+    options.add( feel_options() );
+    options.add_options()
+        ("nb-sensors", po::value<int>()->default_value(4), "nb of sensors in x direction")
+        ("radius", po::value<double>()->default_value(0.025), "radius of sensors")
+        ("trainset-size", po::value<int>()->default_value(10), "number of parameter in the trainset")
+        ( "do-ortho", po::value<bool>()->default_value(false), "")
+        ;
+    options.add(pbdw_options());
+
+    return options;
+}
+
+inline
+AboutData
+makeAbout()
+{
+    AboutData about( "test_pbdw" ,
+                     "test_pbdw" ,
+                     "0.1",
+                     "PBDW test",
+                     Feel::AboutData::License_GPL,
+                     "Copyright (c) 2017 Feel++ Consortium" );
+
+    return about;
+}
+
+bool orthonormalize(std::vector<std::shared_ptr<Pch_element_type<Mesh<Simplex<2>>,1>>>& wn )
```

Comment:
shouldn't it be in the library and be customozed with a scalar product ?

Reply:
it should be in ReducedBasisSpace in my opinion.

Reply:
why not!  you could just have a collection of functions that you want to orthonormalize with respect to a scalar product. ReducedBasisSpace is fine for me. could you add the scalar product customization ?